File: services/identity-service/src/main/java/br/com/deltaglobalbank/identity/domain/token/SigningKeyStatus.java
package br.com.deltaglobalbank.identity.domain.token;

import java.util.Arrays;

public enum SigningKeyStatus {
    ACTIVE,
    RETIRED,
    REVOKED;

    public String toDatabaseValue() {
        return name().toLowerCase();
    }

    public static SigningKeyStatus fromDatabaseValue(String value) {
        return Arrays.stream(values())
            .filter(status -> status.name().equalsIgnoreCase(value))
            .findFirst()
            .orElseThrow(() -> new IllegalArgumentException("signing_key_status_invalid: " + value));
    }
}

File: services/identity-service/src/main/kotlin/br/com/deltaglobalbank/identity/domain/token/SigningKeyStatus.kt
package br.com.deltaglobalbank.identity.domain.token

enum class SigningKeyStatus {
    ACTIVE,
    RETIRED,
    REVOKED;

    fun toDatabaseValue(): String = name.lowercase()

    companion object {
        fun fromDatabaseValue(value: String): SigningKeyStatus =
            entries.firstOrNull { it.name.equals(value, ignoreCase = true) }
                ?: throw IllegalArgumentException("signing_key_status_invalid: $value")
    }
}
